### PR TITLE
Add an example of Chapel using Fortran functions and variables

### PR DESCRIPTION
**Pull request by daviditen**
This example defines some subroutines, functions, and variables in Fortran
and uses them in Chapel using 'extern' prototypes.

Since we don't have a header file with function prototypes in Fortran like
we do in C, we need to generate them ourselves.  Added a new pragma
"generate signature" that can be used on extern functions or variables and
tells codegen to generate prototypes for them even though they're extern.

### CODE REVIEW

**Comment by daviditen** [comment]
Regular Linux64 paratest:
[Summary: #Successes = 9133 | #Failures = 0 | #Futures = 0]

File: test/interop/fortran/ChapelCallFortran/sub_test [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+for path in $libPaths ; do
+  if [[ -f $path/libgfortran.a ]] ; then
+    export LIBGFORTRAN_PATH=$path
+    break
+  fi
+done
+
+$CHPL_HOME/util/test/sub_test $1
+
+rm -f testCallFortran fortranProcs.o testprocs.mod
```
Comment (lydia-duncan) [inline test/interop/fortran/ChapelCallFortran/sub_test]: The .cleanfiles should render this line unnecessary

Comment (daviditen) [reply test/interop/fortran/ChapelCallFortran/sub_test]: The .cleanfiles cleans up at the beginning of the test, but leaves these files sitting around after the test finishes.  This cleans up after the test so it doesn't leave anything behind.

Comment (lydia-duncan) [reply test/interop/fortran/ChapelCallFortran/sub_test]: Ah, right!  Nevermind then!

**Pull request merged**